pkg/apis/kubermatic/v1/ipam_allocation.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// IPAMAllocationResourceName represents "Resource" defined in Kubernetes.
+	IPAMAllocationResourceName = "ipamallocation"
+
+	// IPAMAllocationKindName represents "Kind" defined in Kubernetes.
+	IPAMAllocationKindName = "IPAMAllocation"
+)
+
+// +kubebuilder:object:generate=true
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type="date"
+
+// IPAMAllocation is the object representing the allocation of the
```

Comment:
more something like ... representing an allocation from an IPAMPool made for a particular KKP user cluster.